### ISSUE TITLE
Different behavior of functions toDateTime & toDateTime64

### ISSUE THREAD

**Issue by millin** [ClickHouse/ClickHouse#9781]
**Describe the bug**
Functions toDateTime & toDateTime64 give a different result.

**How to reproduce**
ClickHouse server version 20.3.3 revision 54433.

```
SELECT timezone()

┌─timezone()────┐
│ Europe/Moscow │
└───────────────┘

SELECT toDateTime(today(), 'UTC')                                                                                                                                                                                                                    
                                                                                                                                                                                                                                                     
┌─toDateTime(today(), 'UTC')─┐                                                                                                                                                                                                                       
│        2020-03-20 00:00:00 │                                                                                                                                                                                                                       
└────────────────────────────┘

SELECT toDateTime64(today(), 0, 'UTC')                                                                                                                                                                                                               
                                                                                                                                                                                                                                                     
┌─toDateTime64(today(), 0, 'UTC')─┐                                                                                                                                                                                                                  
│             2020-03-19 21:00:00 │                                                                                                                                                                                                                  
└─────────────────────────────────┘
```
СС @Enmk

**Comment by Enmk** [comment]
Will make sure it is fixed with https://github.com/ClickHouse/ClickHouse/pull/9574

**Comment by Enmk** [comment]
> Will make sure it is fixed with #9574

Ooops, the issue is still there and requires separate fix.

**Comment by alexey-milovidov** [comment]
It's working, I will add a test:

```
SELECT toDate(toDateTime64(today(), 0, 'UTC')) = toDate(toDateTime(today(), 'UTC'))

Query id: 455e479d-16fc-44ca-88dc-6de157360d3f

┌─equals(toDate(toDateTime64(today(), 0, 'UTC')), toDate(toDateTime(today(), 'UTC')))─┐
│                                                                                   1 │
└─────────────────────────────────────────────────────────────────────────────────────┘
```
